lesson02/elb_asg/main.tf

```diff
@@ -0,0 +1,113 @@
+// Configure AWS Cloud provider
+provider "aws" {
+  region = "${var.aws-region}"
+}
+
+# terraform {
+#   backend "s3" {
+#     bucket = "sample-app-terraform"
+#     key    = "terraform/lesson_2.tfstate"
+#     region = "us-east-1"
+#   }
+# }
+
+#--------------------------------------------------------------
+# Launch configuration
+#--------------------------------------------------------------
+resource "aws_launch_configuration" "lc" {
+  lifecycle {
+    create_before_destroy = true
+  }
+  name_prefix                 = "${var.metadata["appname"]}-${var.env}-lc-${var.metadata["appversion"]}-"
+  image_id                    =  "${data.aws_ami.amazon_linux.id}"
+  instance_type               = "${var.instance_type}"
+  security_groups             = ["${aws_security_group.web.id}"]
+  //iam_instance_profile        = "${var.iam_role}" //Optional
+  user_data                   = "${data.template_file.deploy_sh.rendered}"
+  key_name                    = "${var.key_name}"
+  associate_public_ip_address = "${var.associate_public_ip_address}"
+
+}
+
+#===========================================
+#  Auto scaling Group
+#===========================================
+resource "aws_autoscaling_group" "asg" {
+  name_prefix          = "${var.metadata["appname"]}-${var.env}-asg-${var.metadata["appversion"]}-"
+  launch_configuration = "${aws_launch_configuration.lc.name}"
+  availability_zones   =  ["${data.aws_availability_zones.available.zone_ids}"]
+  load_balancers       = ["${aws_elb.elb.id}"]
+  health_check_type    = "${var.health_check_type}"
+  health_check_grace_period = "${var.health_check_grace_period}"
+  default_cooldown = "${var.default_cooldown}"
+  
+  min_size              = "${var.min_size}"
+  max_size              = "${var.max_size}"
+  wait_for_elb_capacity = "${var.min_size}"
+
+  desired_capacity = "${var.desired_capacity}"
+
+  vpc_zone_identifier       = ["${data.aws_subnet_ids.vpc_subnets.ids}"]
+  // autoscaling group metrics as a group
+  #wait_for_capacity_timeout = "20m"
```

Comment:
If this code is not needed, could we delete it?

Reply:
I wanted to leave it there to show the users on how to enable additional metrics for ASG alarms and policies.
Should i Still delete it ?

Reply:
can we add additional description about the commented code?

Reply:
yes

Reply:
please push the code to the branch